### PR TITLE
Asset detail base

### LINKED ISSUE #2977: Base components for asset detail page
Part of epic #2896 

## Is your feature request related to a problem? Please describe 👀
[Figma designs](https://www.figma.com/file/rIK2zVFUVhtbuXo0RBzOzw/0033-2305_omSupply-ColdChain-UI?type=design&node-id=2653-16424&mode=design&t=8KRUuAo7s1V0r0RM-0)

image

image

Comments on the page:
* Heading should be 'locations' plural
* asking if locations can be created here - see the above mock up for that one
* can we add date pickers for the installation and replacement dates?
* is 'reason' free text?
* which fields are required before save is enabled?
* what does 'view catalogue' do? answer: it shows the catalogue details for the asset. perhaps best as a tab to avoid navigating away
* what does the export do? why is it a split button?

## Describe the solution you'd like 🎁

**Navigation:**  
1. … > Asset Register: List View > Click on a row in the list > Asset Register: Detailed View
    … > CCE Inventory: List View > Click on a row in the list > CCE Inventory: Detailed View
2. … > Asset Register: List View > Click on Add asset button > Add asset modal > Asset Register: Detailed View (tab1: Summary)
    … > CCE Inventory: List View > Click on Add CCE button > Add CCE modal > CCE Inventory: Detailed View (tab1: Summary)

**Headers**
1. Breadcrumbs: Asset Register / Asset Identifier
   Breadcrumbs: CCE Inventory / CCE Identifier
2. Buttons:
    1. Add Log Entry (or “Update Status” ?)
    2. View catalogue (?)
    3. Export/Print:
      1. to print the cold chain report
      2. Logs history
      3. Other custom forms (like a “Maintenance Service Sheet”)
3. Three tabs: 
    1. Summary 
    2. Details
    3. Logs
    4. (Documents) ?
        1. Documents from the catalogue (eg. user guides, troubleshooting instructions)
        2. Documents for this asset / CCE (insurance certificate, warranty certificate)

**Update status button**
image

image

**Tab1: Summary Page**
1. Asset identification fields
    1. Category / Type
    2. Make / Model
    3. Serial Number
    4. Installation date / Age
    5. Replacement date (planned)
    6. Photo ?
2. Functional Status:
    1. Functional status: 
        1. Current Functional Status (click on it to create a new log)
        2. Current functional last updated
        3. Current functional reason
    2. 3 most recent logs: 
        1. Type: Maintenance, Temperature Breach
        2. date / user (for cold chain breach, user = sensor’s name)
        3. Previous functional status
        4. New functional status
    3. Upload / Download documents (one separate tab?)
        1. Download documents from the catalogue (user guides, basic troubleshooting instructions)
        2. Upload documents specific to the asset (warranty certificate, insurance certification, invoices, photos…)
    4. Staff: 
        1. Technician (user) + phone number, email
        2. Custodian (user)
3. Cold Chain: 
    1. (Cold) Storage Location(s)
        1. click on it to view/update location
        2. If no location available possibility to create a location for the asset
        3. Sensors (derived from the storage location)
        4. Maybe just a quick link to cold chain details (monitoring page)

**Tab2: Details**
Catalogue details (class and category)
Could be replaced by a link to the catalogue details view ?

**Tab3: Logs**
To see a list of all the asset’s / CCE’s logs (or functional status update)
Asset logs: TBD
CCE logs: [See table for reference](https://docs.google.com/spreadsheets/d/1l1x38KT5cdt4Zd5piMWp3FuBuiSUv4faMHP8utDJDqU/edit#gid=505652337&range=A1)

**Tab4: Documents**

### Moneyworks Jobcode 🧰

`OMSCC:AST`

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #2977

# 👩🏻‍💻 What does this PR do? 
 
Creates the base components for the detail view

Screenshot 2024-03-04 at 5 43 17 PM

I thought to add the 'Update Status' when the api is available, and as a separate PR because that's a bit of work.
Each tab is a separate component, might do them all together though. Then need to wire in the save mutation.

# 🧪 How has/should this change been tested? 

## 📃 Documentation

_No user facing changes_

or

- [ ] Functional change 1
- [ ] Functional change 2

### CODE REVIEW

File: server/repository/src/migrations/v1_08_00/assets/asset.rs [hunk 1]
```diff
@@ -41,5 +41,14 @@ pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
         "#,
     )?;
 
+    if cfg!(feature = "postgres") {
+        sql!(
+            connection,
+            r#"
+            ALTER TYPE context_type ADD VALUE 'ASSET' AFTER 'REPACK';
```
Comment (jmbrunskill) [inline server/repository/src/migrations/v1_08_00/assets/asset.rs:48]:
```suggestion
            ALTER TYPE context_type ADD VALUE 'ASSET' BEFORE 'INBOUND_SHIPMENT';
```
Nit: If you want the order consistent between the postgres and rust?

**Review by jmbrunskill — APPROVED** [review]
Looks good to me!

image

I assume you're expecting no reports to show up at this stage?

Mock up shows a 'View Catalogue' button? Might be worth adding?
image

File: server/repository/src/migrations/v1_08_00/assets/asset.rs (hunk 1 above)
Comment (mark-prins) [reply server/repository/src/migrations/v1_08_00/assets/asset.rs:48]: cool thanks. wasn't sure if the order mattered, but I'd prefer it alphabetical

**Comment by mark-prins** [comment]
> Mock up shows a 'View Catalogue' button? Might be worth adding?

Yeah - I didn't get an answer re functionality so have left it off. my problem is that the button shows the catalogue item, but we don't have enough detail in the catalogue to make this useful. the catalogue has class, category and type. the first is irrelevant ( all are 'cold chain equipment' ) and the other two are already shown in the summary

**Comment by adamdewey** [comment]
> > Mock up shows a 'View Catalogue' button? Might be worth adding?
> 
> Yeah - I didn't get an answer re functionality so have left it off. my problem is that the button shows the catalogue item, but we don't have enough detail in the catalogue to make this useful. the catalogue has class, category and type. the first is irrelevant ( all are 'cold chain equipment' ) and the other two are already shown in the summary
> 
> > I assume you're expecting no reports to show up at this stage?
> 
> Yes, will need to implement at some point. Will investigate that one further, the reports come through from mSupply and I'll need to see what's required for that.

Sorry, meant to follow this up - yeah I don't think we need the `View Catalogue` button in the register at the moment - most users won't have access the catalogue and they can just go to the side menu if they do?

What is the easiest way for a user to trace back an individual asset in the register to its parent catalogue entry? Do they have have to search by name and model? (i.e. there's no catalogue reference number?)

**Comment by mark-prins** [comment]
We could bring that through on one of the tabs - honestly though, there's so little in the catalogue there's not much value in it. perhaps we should be importing a lot more from the catalogue? might be best if we show you where things are at and you can advice what's missing?

**Comment by adamdewey** [comment]
> 
We could bring that through on one of the tabs - honestly though, there's so little in the catalogue there's not much value in it. perhaps we should be importing a lot more from the catalogue? might be best if we show you where things are at and you can advice what's missing?

Ah, you mean importing from the WHO PQS catalogue? 

Yeah I think we would need to get quite a lot of info from it eventually as the (likely upcoming) gaps analysis work will use a lot of that PQS data to do its analysis. 

But yeah let's catch up next week!

**Pull request merged**